modules/indexing/src/test/java/org/apache/ignite/cache/query/MultifieldIndexQueryTest.java

```diff
@@ -0,0 +1,598 @@
+        ignite = startGrids(nodesCnt);
+
+        cache = ignite.cache(CACHE);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() {
+        stopAllGrids();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        CacheConfiguration<?, ?> ccfg = new CacheConfiguration<>()
+            .setName("TEST_CACHE")
+            .setIndexedTypes(Long.class, Person.class)
+            .setQueryParallelism(1);
+
+        cfg.setCacheConfiguration(ccfg);
+
+        return cfg;
+    }
+
+    /** */
+    @Test
+    public void testQueryKeyPKIndex() {
+        insertData();
+
+        int pivot = new Random().nextInt(CNT);
+
+        IndexQuery<Long, Person> qry = new IndexQuery<Long, Person>(Person.class, "_key_PK")
+            .setCriteria(lt("_KEY", (long) pivot));
+
+        checkPerson(cache.query(qry), 0, pivot);
+    }
+
+    /** */
+    @Test
+    public void testEmptyCacheQuery() {
+        IndexQuery<Long, Person> qry = new IndexQuery<Long, Person>(Person.class)
+            .setCriteria(lt("id", Integer.MAX_VALUE), lt("secId", Integer.MAX_VALUE));
+
+        QueryCursor<Cache.Entry<Long, Person>> cursor = cache.query(qry);
+
+        assertTrue(cursor.getAll().isEmpty());
+
+        // Check same query with specify index name.
+        qry = new IndexQuery<Long, Person>(Person.class, INDEX)
+            .setCriteria(lt("id", Integer.MAX_VALUE), lt("secId", Integer.MAX_VALUE));
+
+        assertTrue(cache.query(qry).getAll().isEmpty());
+    }
+
+    /** */
+    @Test
+    public void testCheckBoundaries() {
+        cache.put(1L, new Person(0, 1));
+        cache.put(2L, new Person(1, 0));
+        cache.put(3L, new Person(1, 1));
+
+        IndexQuery<Long, Person> qry = new IndexQuery<Long, Person>(Person.class)
+            .setCriteria(between("id", 0, 1), eq("secId", 1));
+
+        List<Cache.Entry<Long, Person>> result = cache.query(qry).getAll();
+
+        assertEquals(2, result.size());
+
+        result.sort(Comparator.comparingLong(Cache.Entry::getKey));
+
+        assertEquals(1L, (long) result.get(0).getKey());
+        assertEquals(3L, (long) result.get(1).getKey());
+
+        assertEquals(new Person(0, 1), result.get(0).getValue());
+        assertEquals(new Person(1, 1), result.get(1).getValue());
+    }
+
+    /** */
+    @Test
+    public void testLtQueryMultipleField() {
+        insertData();
+
+        int pivot = new Random().nextInt(CNT);
+
+        // Should return empty result for ID that less any inserted.
+        IndexQuery<Long, Person> qry = new IndexQuery<Long, Person>(Person.class)
+            .setCriteria(lt("id", -1), lt("secId", pivot));
+
+        assertTrue(cache.query(qry).getAll().isEmpty());
+
+        // Should return all data for ID and SECID that greater any inserted.
```

Comment:
Don't understand this comment (and the following comments in this method). Why `greater` if the criteria are `lt`?

Reply:
Greater is related to specified value in query. We set ID = 1 that greater that any inserted (id = 0 for all rows).